### ISSUE TITLE
Cache documentation issue w.r.t `generateFunc`

### ISSUE THREAD

**Issue by vladikoff** [hapijs/hapi#4079]
#### Context

* *node version*:  12
* *module version*:  hapi 17 , catbox-memory 5 , catbox-redis 5

#### What are you trying to achieve or the steps to reproduce ?

The docs under https://hapi.dev/api/?v=19.1.1#-servermethodname-method-options state the following: 

> cache - the same cache configuration used in server.cache(). The generateTimeout option is required.

The `server.cache` docs mention the `generateFunc` option. However looking at the code `generateFunc` must not be set on the server.method:

![image](https://user-images.githubusercontent.com/128755/77456664-ec673280-6dd1-11ea-9383-4f8259ad0a72.png)

tl;dr `server.method` docs should mention that the `generateFunc` is not supported in `serverm.method` options.
